bindings/matrix-sdk-crypto-ffi/src/logger.rs

```diff
@@ -21,11 +21,23 @@ pub trait Logger: Send {
     /// Called every time the Rust side wants to post an error log line.
     fn log_error(&self, message: String, data: String);
 }
-
+pub struct LoggerWrapper {
+    inner: Arc<Mutex<Box<dyn Logger>>>,
+    level: tracing_core::Level,
+}
 impl Write for LoggerWrapper {
     fn write(&mut self, buf: &[u8]) -> Result<usize> {
-        let data = String::from_utf8_lossy(buf).to_string();
-        self.inner.lock().unwrap().log(data);
+        let message = String::from_utf8_lossy(buf).to_string();
+  match self.level {
+            tracing_core::Level::ERROR => self.inner.lock().unwrap().log_error(message, String::new()),
```

Comment:
You may want to write `use tracing_core::Level` at the top of the file, so that it makes the code shorter.